src/data/vocab_base.h

```diff
@@ -34,6 +53,20 @@ class IVocab {
 
   virtual std::string decode(const Words& sentence,
                              bool ignoreEos = true) const = 0;
+
+  virtual void encode(sentencepiece::util::min_string_view /*textSpan*/, // span of text to be encoded
+                      sentencepiece::SentencePieceText* /*encoded*/) const {
```

Comment:
And it won't ever be because of the wrong class.  Also, now it won't compile without sentencepiece.